Frameworks/Blockcerts/CertificateV3.swift

```diff
@@ -0,0 +1,264 @@
+                        introductionMethod: .unknown,
+                        analyticsURL: nil)
+    }
+    
+    static func parse(recipientJSON json: [String : Any]) -> Recipient? {
+        if let recipientData = json["credentialSubject"] as? [String : Any] {
+            let publicKey = recipientData["publicKey"] as? BlockchainAddress
+            let name : String = recipientData["name"] as? String ?? recipientData["id"] as! String
+            
+            return Recipient(name: name,
+                             publicAddress: publicKey)
+        }
+        
+        return nil
+    }
+    
+    static func parse(metadataJSON: [String : Any]) -> Metadata? {
+        if let metadataString = metadataJSON["metadata"] as? String {
+            do {
+                let data = metadataString.data(using: .utf8)
+                let metadataJson : [String : Any] = try JSONSerialization.jsonObject(with: data!, options: []) as! [String: Any]
+                return Metadata(json: metadataJson)
+            } catch {
+                print("Failed to parse metadata json:")
+                print(metadataString)
+            }
+        }
+        
+        return nil
+    }
+    
+    static func parse(assertionJSON: AnyObject?) -> Assertion? {
+        guard let assertionData = assertionJSON as? [String : Any],
+            let issuedOnString = assertionData["issuanceDate"] as? String,
+            let issuedOnDate = issuedOnString.toDate(),
+            let assertionID = assertionData["id"] as? String else {
+                return nil
+        }
+        
+        var assertionUID : String!
+        let assertionIDURL = URL(string: assertionID)
+        if assertionIDURL != nil {
+            assertionUID = assertionID
+        } else if let range = assertionID.range(of:Constants.guidRegexp, options: .regularExpression) {
+            assertionUID = String(assertionID[range])
+        }
+        
+        guard assertionUID != nil else {
+            return nil
+        }
+        
+        var metadataJson : [String : Any] = [:]
+        if let metadataString = assertionData["metadataJson"] as? String {
+            do {
+                let data = metadataString.data(using: .utf8)
+                metadataJson = try JSONSerialization.jsonObject(with: data!, options: []) as! [String: Any]
+            } catch {
+                print("Failed to parse metadata json:")
+                print(metadataString)
+            }
+        }
+        
+        let htmlDisplay = assertionData["displayHtml"] as? String
+        
+        return Assertion(issuedOn: issuedOnDate,
+                         uid: assertionUID,
+                         id: assertionIDURL,
+                         metadata: Metadata(json: metadataJson),
+                         htmlDisplay: htmlDisplay)
+    }
+    
+    static func parse(verifyJSON: [String: AnyObject?]) -> Verify? {
+        guard let verifyData = verifyJSON as? [String : AnyObject],
```

Comment:
Nit: verifyJSON is already  [String: AnyObject?]. The Casting is redundant.

Reply:
👍